bobble:
    bobble_balance_controller:
      type: bobble_controllers/BobbleBalanceController
      joints:
        - right_wheel_hinge
        - left_wheel_hinge
      InSim: false
      StartingTiltSafetyLimitDegrees: 8.0
      MaxTiltSafetyLimitDegrees: 20.0
      MotorEffortMax: 0.2
      MotorEffortToTorqueSimFactor: 1.0 # Should be set to 1.0 for HW
      WheelVelocityAdjustment: 0.05
      MadgwickFilterGain: 0.20
      MeasuredTiltFilterGain: 0.15
      MeasuredTiltDotFilterGain: 0.15
      MeasuredHeadingFilterGain: 0.5
      MeasuredTurnRateFilterGain: 0.1
      LeftWheelVelocityFilterGain: 0.25
      RightWheelVelocityFilterGain: 0.25
      DesiredForwardVelocityFilterGain: 0.25
      DesiredTurnRateFilterGain: 0.25
      MaxVelocityCmd: 0.75
      VelocityCmdScale: 10.0
      TurnCmdScale: 30.0
      VelocityControlKp: 0.095
      VelocityControlKi: 0.000
      VelocityControlAlphaFilter: 0.45
      VelocityControlMaxIntegralOutput: 0.010
      VelocityControlOutputLimitDegrees: 12.0
      TiltControlKp: 1.0
      TiltControlKd: 0.05
      TiltControlAlphaFilter: 0.05
      TiltOffset: 0.10
      TiltDotOffset: -0.1457
      RollDotOffset: 0.0
      YawDotOffset: -0.0137
      TurningControlKp: 0.065
      TurningControlKi: 0.0
      TurningControlKd: 0.0
      TurningOutputFilter: 0.5

